pkg/api/course.go

```diff
@@ -0,0 +1,140 @@
+package api
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/tonymontanapaffpaff/golang-training-university/pkg/data"
+
+	"github.com/gorilla/mux"
+)
+
+type courseAPI struct {
+	data *data.CourseData
+}
+
+func ServeCourseResource(r *mux.Router, data data.CourseData) {
+	api := &courseAPI{data: &data}
+	r.HandleFunc("/courses", api.getAllCourses).Methods("GET")
+	r.HandleFunc("/courses/{code}", api.getCourse).Methods("GET")
+	r.HandleFunc("/courses", api.createCourse).Methods("POST")
+	r.HandleFunc("/courses/{code}", api.updateCourseDescription).Methods("PATCH")
+	r.HandleFunc("/courses/{code}", api.deleteCourse).Methods("DELETE")
+}
+
+func (a courseAPI) getAllCourses(writer http.ResponseWriter, request *http.Request) {
+	courses, err := a.data.ReadAll()
+	if err != nil {
+		_, err := writer.Write([]byte("got an error when tried to get list of courses"))
+		if err != nil {
+			log.Println(err)
+		}
+		return
```

Comment:
forgot to added error to header in the response, before return should be `writer.WriteHeader(http.StatusInternalServerError)`